x/dex/utils/math.go

```diff
@@ -66,3 +66,25 @@ func MustSafeUint64ToInt64(in uint64) (out int64) {
 
 	return safeInt64
 }
+
+func Log(val, base math_utils.PrecDec) (math_utils.PrecDec, error) {
+	valueFloat64, err := val.Float64()
```

Comment:
1. Given that I am writing this as a general method it is possible for a preDec to overflow a float64. I prefer to keep it as is. 
2. Generally speaking yes, but for the specific case I have written a bunch of tests to ensure that within ticklimits there is not loss precision

Reply:
> 1. Given that I am writing this as a general method it is possible for a preDec to overflow a float64. I prefer to keep it as is.

Actually it can not, float64 a way wider range. It silently truncates original number keeps only precise number of digits (and loose precision), but it does not throw any error. That the reason why i suggest MustFloat64 (not silently ignore an error), it saves you a couple of lines of unnecessary error checks